net/https-dns-proxy/files/https-dns-proxy.init

```diff
@@ -56,19 +46,34 @@ start_instance() {
 	append_parm "$cfg" 'proxy_server' '-t'
 	append_parm "$cfg" 'logfile' '-l'
 	append_bool "$cfg" 'use_http1' '-x'
-	append_match "$cfg" 'verbosity' '-v' '1'
-	append_match "$cfg" 'verbosity' '-vv' '2'
-	append_match "$cfg" 'verbosity' '-vvv' '3'
-	append_match "$cfg" 'verbosity' '-vvvv' '4'
+	append_bool "$cfg" 'ipv4_resolvers' '-4'
+
+	config_get verbosity "$cfg" 'verbosity' "0"
+	for i in $(seq 1 $verbosity); do
+		xappend "-v"
+	done
 
 	procd_open_instance
 # shellcheck disable=SC2086
-	procd_set_param command ${PROG} -4 ${param}
+	procd_set_param command ${PROG} ${param}
+	procd_set_param stderr 1
+	procd_set_param stdout 1
 	procd_set_param respawn
 	procd_close_instance
 
 	config_get listen_addr "$cfg" 'listen_addr' '127.0.0.1'
 	config_get listen_port "$cfg" 'listen_port' "$p"
+
+	# Don't add the any address to dnsmasq
+	case $listen_addr in
+		0.0.0.0|::ffff:0.0.0.0)
+			listen_addr='127.0.0.1'
+			;;
+		::)
+			listen_addr='::1'
+			;;
+	esac
```

Comment:
Dnsmasq won't work if it forwards to `::` or `0.0.0.0`, [but it can be useful to bind a service to them](https://github.com/openwrt/openwrt/blob/master/package/network/services/uhttpd/files/uhttpd.config#L5).  
So to satiate both we override those special cases to forward to localhost.

Reply:
Do any other resolving proxies for OpenWrt do that? I'm not strongly opposed to the change, I'm just questioning its practicality for not web server.

Reply:
It's more of a socket concept, rather than a server specific concept.  So I would assume that others at least facilitate it like `https_dns_proxy`, though I am not aware of others specifically.

Reply:
I'm not sure if this is a good feature for the proxy. It already substitutes missing listen address for the localhost.
In case of the web/ssh server zero-address could be useful as a wildcard instead of binding the service to a specific interface, I'm not sure if a special handling should be done for this proxy.
@hnyman, @jow-, @aarond10 care to weigh in?

Reply:
What if I'm not using dnsmasq? Or I'm running it on another device? It seems confusing to me that the listen address specified does not get used. I may be missing something.

Reply:
If you care for my use case. I have services running on different subnets that need to bypass dnsmasq, so localhost isn't an option. I could technically bind to one of the subnet adapter's, but then I'd have to A) poke holes in the firewall to allow data crossing subnets that are intended for isolation or B) run a copy of https-dns-proxy for each subnet. Binding to the any address works for me because dnsmasq and local services can still use localhost and my other subnet services (docker images) and clients can use either dnsmasq (with cache and redirection) or https-dns-proxy (upstream) depending on their needs.

Reply:
I'm still a bit confused I think. Can't you explicitly set `option listen_addr '127.0.0.1'`? If the init script silently replaces the wildcard address with localhost, it's going to break anyone that actually wants `options listen_addr '0.0.0.0'`, right?

Reply:
The `listen_addr` option, when combined with `update_dnsmasq_config`, has two purposes. The address the proxy listens on and the address that dnsmasq uses as the upstream. Currently, we bind to an address and then configure dnsmasq to forward to the same address. However, this breaks if we bind to [an any address](https://en.wikipedia.org/wiki/0.0.0.0) as we are listening on all addresses but we are configuring dnsmasq to forward to the default route and not the proxy like the previous cases. This change just makes it so the behaviour of listening to the specified address and forwarding dnsmasq to the proxy the same regardless of which address we listen to.

Reply:
Arguably, dnsmasq forwarding to the default route could be valid, but it would be different behaviour that I think would be best suited via disabling `update_dnsmasq_config` or at least a separate option.

Reply:
Oh, if this code is in place just for the benefit of dnsmasq, should it not be moved to `dnsmasq_add_doh_server`?

Reply:
moved as suggested, I also modified the arguments of `dnsmasq_add_doh_server` to keep the address stripping simple.